core/controller/src/main/scala/org/apache/openwhisk/core/controller/Actions.scala

```diff
@@ -624,6 +626,30 @@ trait WhiskActionsApi extends WhiskCollectionAPI with PostActionActivation with
     })
   }
 
+  private def checkNamespaceAndSystemLimits(user: Identity, content: WhiskActionPut)(
```

Comment:
Should this be two separate functions? The context in which you check the set limits for a namespace are different from the context you check the system limits. You check against the system limits when creating the action or updating your namespace's configuration. When you check against the namespace's limit it would be on the execution path when an activation is occurring.

Reply:
Because updating the namespace limit configuration directly inserts/updates into the Couch DB, I've updated the code to fallback to the system limit if the namespace limit that exceeds the system limit.

So, namespace limits that can not exceed the system limit are checked when an action is created/invoked.